File: programmers/Lv1/모의고사.py
"""
    lv.1 모의고사

    1.  나올 수 있는 답의 가지수가 정해져 있으므로 완전 탐색을 진행
"""


def solution(answers):
    answer = []

    score = {1: 0, 2: 0, 3: 0}

    mark1 = [1, 2, 3, 4, 5]
    mark2 = [2, 1, 2, 3, 2, 4, 2, 5]
    mark3 = [3, 3, 1, 1, 2, 2, 4, 4, 5, 5]

    for i, v in enumerate(answers):
        if v == mark1[i % len(mark1)]:
            score[1] += 1
        if v == mark2[i % len(mark2)]:
            score[2] += 1
        if v == mark3[i % len(mark3)]:
            score[3] += 1

    maxScore = max(list(score.values()))
    answerTuple = list(filter(lambda x: x[1] == maxScore, list(score.items())))
    answer = list(map(lambda x: x[0], answerTuple))

    return answer


result = solution([1, 3, 2, 4, 2])
print(result)

File: programmers/Lv1/모의고사.js
/* 
lv.1 모의고사
 */

function solution(answers) {
  var answer = [];

  const scores = [0, 0, 0];
  const case1 = [1, 2, 3, 4, 5];
  const case2 = [2, 1, 2, 3, 2, 4, 2, 5];
  const case3 = [3, 3, 1, 1, 2, 2, 4, 4, 5, 5];

  for (let i = 0; i < answers.length; i++) {
    if (answers[i] === case1[i % case1.length]) {
      scores[0]++;
    }
    if (answers[i] === case2[i % case2.length]) {
      scores[1]++;
    }
    if (answers[i] === case3[i % case3.length]) {
      scores[2]++;
    }
  }

  const maxScore = Math.max(...scores);

  scores.forEach((v, i) => {
    if (v === maxScore) {
      answer.push(i + 1);
    }
  });

  return answer;
}

const answers = [1, 3, 2, 4, 2];

console.log(solution(answers));
